catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/RemoveAllOfflineSolrGazetteerCommand.java

```diff
@@ -0,0 +1,91 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.solr.offlinegazetteer;
+
+import com.google.common.collect.ImmutableList;
+import java.util.concurrent.TimeUnit;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.client.solrj.UnavailableSolrException;
+import org.codice.solr.factory.SolrClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+@Command(
+  scope = "offline-solr-gazetteer",
+  name = "removeall",
+  description = "Deletes all items in the offline solr gazetteer solr core"
+)
+public class RemoveAllOfflineSolrGazetteerCommand implements Action {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(RemoveAllOfflineSolrGazetteerCommand.class);
+
+  @Reference protected Session session;
+
+  @Reference private SolrClientFactory clientFactory;
+
+  @Option(
+    name = "--force",
+    aliases = {"-f"},
+    description = "Force the removal without a confirmation message."
+  )
+  boolean force = false;
+
+  private final RetryPolicy retryPolicy =
+      new RetryPolicy()
+          .retryOn(ImmutableList.of(UnavailableSolrException.class, SolrServerException.class))
+          .withMaxDuration(5, TimeUnit.SECONDS)
+          .withBackoff(25, 1_000, TimeUnit.MILLISECONDS);
+
+  @Override
+  public Object execute() throws Exception {
+    if (!force) {
+      String answer =
+          session
+              .readLine(
+                  "Are you sure you want to remove all gazetteer entries inside of the offline solr gazetteer core?(y/n)",
+                  ' ')
+              .toLowerCase();
+      if (!("y".equals(answer) || "yes".equals(answer))) {
+        session.getConsole().println("Aborting.");
+        return null;
+      }
+    }
+    try {
+      SolrClient solrClient =
+          clientFactory.newClient(OfflineGazetteerPlugin.STANDALONE_GAZETTEER_CORE_NAME);
+
+      Failsafe.with(retryPolicy).get(() -> solrClient.deleteByQuery("*:*"));
```

Comment:
@paouelle may be interested in this -- It seems that with this karaf command (and the other here) if i attempted to use the solrClient returned by `clientFactory.newClient` too *'quickly'* , it would throw a SolrServerException because the client had not connected yet. Hence, I added the retry to have it retry a few times to give it a chance to connect (which did seem to alleviate the issue).

Reply:
i have seen this as well on splitcatalogprovider. should newClient bake in the logic to check/wait with timeout before returning?

Reply:
probably not - there are api methods available on the client itself for either sync or async behavior - see above links.